File: 2646-kth-largest-sum-in-a-binary-tree/2646-kth-largest-sum-in-a-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

from collections import deque
import heapq

class Solution(object):
    def kthLargestLevelSum(self, root, k):
        pq = []
        q = deque([root])

        while q:
            n = len(q)
            level_sum = 0
            for i in range(n):
                node = q.popleft()
                level_sum += node.val

                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)
                
            heapq.heappush(pq, level_sum)
            if len(pq) > k:
                heapq.heappop(pq)
            
        if len(pq) < k:
            return -1
        
        return heapq.heappop(pq)

        

File: 2646-kth-largest-sum-in-a-binary-tree/2646-kth-largest-sum-in-a-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} k
 * @return {number}
 */
var kthLargestLevelSum = function(root, k) {
    let pq = []; 
    let q = [root]; 

    while (q.length > 0) {
        let n = q.length;
        let sum = 0;

        for (let i = 0; i < n; i++) {
            let node = q.shift();
            sum += node.val;

            if (node.left) q.push(node.left);
            if (node.right) q.push(node.right);
        }

        pq.push(sum);
    }

    if(pq.length < k) return -1;

    pq.sort((a, b) => b - a);
    return pq[k-1];
};